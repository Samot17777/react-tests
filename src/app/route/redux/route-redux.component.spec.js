import React from 'react';
import configureStore from 'redux-mock-store'
import {render, shallow, mount} from 'enzyme';
import {Provider} from 'react-redux';
import {RouteReduxComponent} from "./route-redux.component";
import {RouteReduxContainer} from "./route-redux.container";

describe('Route redux component', () => {
    it('renders correctly', () => {
        const props = {
            example: {
                exampleNumber: 1
            }
        };

        const wrapper = render(<RouteReduxComponent {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('changes props correctly', () => {
        const props = {
            example: {
                exampleNumber: 1
            }
        };

        const wrapper = shallow(<RouteReduxComponent {...props}/>);

        expect(wrapper.text()).toBe("1");
        wrapper.setProps({
            example: {
                exampleNumber: 2
            }
        });

        expect(wrapper.text()).toBe("2");
    });
});

const mockStore = configureStore();
describe('Route redux container', () => {
    it('renders correctly', () => {
        const store = mockStore({
            example: {
                exampleNumber: 1
            }
        });

        const wrapper = render(
            <Provider store={store}>
                <RouteReduxContainer/>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('changes props correctly', () => {
        const store = mockStore({
            example: {
                exampleNumber: 1
            }
        });

        // const wrapper = shallow(
        //     <Provider store={store}>
        //         <RouteReduxContainer/>
        //     </Provider>
        // );

        const wrapper = mount(
            <Provider store={store}>
                <RouteReduxContainer/>
            </Provider>
        );

        expect(wrapper.text()).toBe("1");
    });
});