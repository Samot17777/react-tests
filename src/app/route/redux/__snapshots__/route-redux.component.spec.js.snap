// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route redux component renders correctly 1`] = `
<div>
  1
</div>
`;

exports[`Route redux container renders correctly 1`] = `
<div>
  1
</div>
`;
